src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -121,6 +121,28 @@ export class InstallAction extends Action {
 	}
 
 	private install(extension: IExtension): TPromise<void> {
+		// {{SQL CARBON EDIT}}
+		if (this.extension.publisher !== 'Microsoft') {
```

Comment:
I'm assuming this is more of a PM ask? from a security PoV this adds nothing.

Reply:
That's a good point. Is the extension publisher here coming from the extension itself or from our extension gallery JSON file? For this to be secure it needs to be the publisher in the gallery file, and we should prompt for any extensions that aren't in the gallery regardless of publisher.